### ISSUE TITLE
[Xcodeproj] Unknown object version (60).

### ISSUE THREAD

**Issue by mcgrawhill** [fastlane/fastlane#21653]
**New Issue Checklist**

- [X]  Updated fastlane to the latest version
- [X]  I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X]  I read [docs.fastlane.tools](https://docs.fastlane.tools/)
- [X]  I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

**Issue Description**
Fastlane fails bulding MacOS app if **project.pbxproj**  version  is "**60**"

**Command executed**
fastlane with lane desa_app
```
    lane :desa_app do
        build_app(scheme: "XXX", output_directory: XXX, export_method:"developer-id",export_options: {
                  provisioningProfiles: {
                  "com.a.b" => "DeveloperID App Profile",
                  "com.a.b" => "DeveloperID Extension Profile"
                  }
                  })
```

 fastlane finished with errors
[Xcodeproj] Unknown object version (60).

This error only happens when using String Catalog
https://developer.apple.com/documentation/Xcode/localizing-and-varying-text-with-a-string-catalog

project.pbxproj file is updated from version 56 to 60 when enabling string catalog

**Environment**
- fastlane 2.217.0
- macOS 14.1 (23B74)
- xCode 15.0.1 (15A507)

**Complete output when running fastlane, including the stack trace and command used**
```
/Library/Ruby/Gems/2.6.0/gems/xcodeproj-1.22.0/lib/xcodeproj/project.rb:228:in `initialize_from_file': [!] [Xcodeproj] Unknown object version (60). (RuntimeError)
	from /Library/Ruby/Gems/2.6.0/gems/xcodeproj-1.22.0/lib/xcodeproj/project.rb:113:in `open'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane_core/lib/fastlane_core/project.rb:120:in `project'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane_core/lib/fastlane_core/project.rb:207:in `configurations'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/gym/lib/gym/detect_values.rb:163:in `detect_configuration'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/gym/lib/gym/detect_values.rb:34:in `set_additional_default_values'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/gym/lib/gym/module.rb:15:in `config='
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/gym/lib/gym/manager.rb:8:in `work'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/actions/build_app.rb:68:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:134:in `block (2 levels) in parsing_binding'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/lane.rb:41:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.217.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by mcgrawhill** [comment]
Fixed after updating coapods and ruby to last version

**Comment by raghuveeranup** [comment]
@mcgrawhill  
hi...
what do you mean "updating cocoapods and ruby to last version" did you mean the latest version? can you pls specify the version you used. I am facing similar issue when running fastlane in Jenkins.

Issue:

[!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx"} for the 'PBXFileReference' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new
[!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new
[!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new
[!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa.
If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new
+------------------+-----------+
|         Lane Context         |
+------------------+-----------+
| DEFAULT_PLATFORM | ios       |
| BUILD_NUMBER     | 235       |
| PLATFORM_NAME    | ios       |
| LANE_NAME        | ios build |
+------------------+-----------+
**[11:37:32]: [Xcodeproj] Unknown object version (60).**

**Comment by crico** [comment]
Same here. What are the versions that have worked for you? I'm using cocoapods 1.14.3 and ruby 3.1.4

**Comment by vdshko** [comment]
I faced the same issue and the fix for me was just updating the Fastlane version in my Gemfile from 2.214 to 2.219.

**Comment by ansonngch-Sephora** [comment]
I'm facing similar issue in 2.219 :/

**Comment by jojost1** [comment]
I have the same issue but with `[Xcodeproj] Unknown object version (63).` in Xcode 15.3.

**Comment by jamesyorke** [comment]
> I have the same issue but with `[Xcodeproj] Unknown object version (63).` in Xcode 15.3.

Same here

**Comment by raghuveeranup** [comment]
> @mcgrawhill hi... what do you mean "updating cocoapods and ruby to last version" did you mean the latest version? can you pls specify the version you used. I am facing similar issue when running fastlane in Jenkins.
> 
> **Issue**:
> 
> [!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx"} for the 'PBXFileReference' isa. If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new [!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa. If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new [!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa. If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new [!] Xcodeproj doesn't know about the following attributes {"expectedSignature"=>"AppleDeveloperProgram:8VDxxx""} for the 'PBXFileReference' isa. If this attribute was generated by Xcode please file an issue: https://github.com/CocoaPods/Xcodeproj/issues/new +------------------+-----------+ | Lane Context | +------------------+-----------+ | DEFAULT_PLATFORM | ios | | BUILD_NUMBER | 235 | | PLATFORM_NAME | ios | | LANE_NAME | ios build | +------------------+-----------+ **[11:37:32]: [Xcodeproj] Unknown object version (60).**

Resolved after updating to latest Fastlane version.

**Comment by dnhyde** [comment]
I solved it using `XCode 14.0` instead of `XCode 15.3` in the xcode project inspectors (right panel) -> Project Document -> Project Format

|  Issue  |  Fix  |
|    ---   |   ---   |
|  <img width="259" alt="ScreenshotXCode15" src="https://github.com/user-attachments/assets/dc70f059-a8d5-4c72-9ff0-e3e0769c330c">    |     <img width="255" alt="ScreenshotXCode14" src="https://github.com/user-attachments/assets/0d8f0e8c-cc69-42ad-bbd1-cfcb74eadc55">    |

```diff
	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 63;
+	objectVersion = 56;
 	objects = {
```
```diff
 			buildConfigurationList = ... /* Build configuration list for PBXProject "yourProject" */;
-			compatibilityVersion = "Xcode 15.3";
+			compatibilityVersion = "Xcode 14.0";
 			developmentRegion = en;
```
